### PR TITLE
feat(kvm-html): kvm html in ipmi tab

### PR DESCRIPTION
**Pull request by varun257**
### Description of the Change

add button to support kvm html5 url
add url button which opens a new tab
add api calls to support this

### Benefits

Helps user launch kvm in browser

### Possible Drawbacks

None

### Applicable Issues

resolves MANAGER-1368

### CODE REVIEW

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.html [hunk 1]
```diff
@@ -110,6 +110,25 @@ <h3 data-translate="server_configuration_kvm_informations"></h3>
                                     data-translate="server_configuration_impi_java_download">
                             </button>
                         </li>
+                        <li ng-if="ipmi.model.supportedFeatures.kvmipHtml5URL">
```
Comment (antleblanc) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.html]:
```suggestion
                        <li data-ng-if="ipmi.model.supportedFeatures.kvmipHtml5URL">
```

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 2]
```diff
@@ -244,6 +243,60 @@ angular.module('App').controller('ImpiCtrl', ($scope, $translate, Server, Pollin
...
+  $scope.getIpmiKvmUrl = function () {
+    $scope.loader.buttonStart = true;
+    $scope.loader.kvmhtmlLoading = true;
+    Server.ipmiStartConnection({
+      serviceName: $stateParams.productId,
+      type: 'kvmipHtml5URL',
+      ttl: $scope.ttl,
+      ipToAllow: $scope.ipmi.model.clientIp,
+    }).then(
+      (task) => {
```
Comment (JeremyDec) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]:
```suggestion
      ({ taskId }) => {
        startIpmiKvmUrlPoll({ id: taskId });
      }
```

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 3]
```diff
@@ -244,6 +243,60 @@ angular.module('App').controller('ImpiCtrl', ($scope, $translate, Server, Pollin
...
+        startIpmiKvmUrlPoll({ id: task.taskId });
+      },
+      (err) => {
```
Comment (JeremyDec) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]:
```suggestion
      ({ data }) => {
      [...]
      Alerter.alertFromSWS($translate.instant('server_configuration_impi_navigation_error'), data, $scope.alert);
```

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 4]
```diff
@@ -244,6 +243,60 @@ angular.module('App').controller('ImpiCtrl', ($scope, $translate, Server, Pollin
...
+  function getKvmUrl() {
+    Server.ipmiGetConnection($stateParams.productId, 'kvmipHtml5URL').then(
+      (data) => {
+        $scope.loader.kvmUrlReady = true;
+        $scope.loader.kvmhtmlLoading = false;
+        $scope.loader.buttonStart = false;
+        $scope.loader.kvmUrl = data.value;
+        Alerter.alertFromSWS($translate.instant('server_configuration_impi_navigation_success'), true, $scope.alert);
+      },
+      (err) => {
```
Comment (JeremyDec) [inline client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]: Same than : https://github.com/ovh-ux/ovh-manager-dedicated/pull/734/files#r227249605

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 5]
```diff
@@ -236,10 +235,64 @@ angular.module('App').controller('ImpiCtrl', ($scope, $translate, Server, Pollin
...
-        Alerter.alertFromSWS($translate.instant('server_configuration_impi_navigation_error'), data.data, $scope.alert);
+        Alerter.alertFromSWS($translate.instant('server_configuration_impi_navigation_error'), data, $scope.alert);
+      },
+    );
+  }
+
+  $scope.getIpmiKvmUrl = function () {
+    $scope.loader.buttonStart = true;
+    $scope.loader.kvmhtmlLoading = true;
+    Server.ipmiStartConnection({
```
[the remark this answers is not in the document]
Comment (varun257) [reply client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]: Not using any return value here. So not returning anything

File: client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js [hunk 6]
```diff
@@ -236,10 +235,64 @@ angular.module('App').controller('ImpiCtrl', ($scope, $translate, Server, Pollin
...
+      serviceName: $stateParams.productId,
+      type: 'kvmipHtml5URL',
+      ttl: $scope.ttl,
+      ipToAllow: $scope.ipmi.model.clientIp,
+    }).then(
+      ({ taskId }) => {
+        startIpmiKvmUrlPoll({ id: taskId });
+      },
+      ({ data }) => {
```
[the remark this answers is not in the document]
Comment (varun257) [reply client/app/dedicated/server/ipmi/dedicated-server-ipmi.controller.js]: @FredericEspiau catch will not work in this. Are you suggesting a restructure of `Promise.then().catch()` ?

**Pull request closed without merging**